### PR TITLE
verify: verify all N signatures of transaction

### LINKED ISSUE #1059: verify: verify all N signatures of transaction

### PR DESCRIPTION
**Pull request by 0x0ece**
This PR verifies all signatures in a tx.

We added an interface for batch verification (batch signatures over a single message), but the implementation is the naive one. Efficient implementation will be done in an independent PR.

Fixes #1059.

This includes tests for #1059 and #1068.

### CODE REVIEW

File: src/ballet/txn/fd_txn.h [hunk 1]
```diff
@@ -57,12 +57,15 @@
...
 
+   FD_TXN_ACTUAL_SIG_MAX: used to allocate arrays of signatures, pubkeys, etc.
+
    From the spec: "The Solana runtime verifies that the number of signatures
    [stored as a compact-u16] matches the number in the first 8 bits of the
    message header."
    Thus this value must live in the range where compact-u16 and uint8
    representations are identical, hence a max of 127. */
 #define FD_TXN_SIG_MAX               (127UL)
+#define FD_TXN_ACTUAL_SIG_MAX        (12UL)
```
Comment (mmcgee-jump) [inline src/ballet/txn/fd_txn.h:68]: Where does 12UL come from?

File: src/ballet/ed25519/fd_ed25519.h [hunk 2]
```diff
@@ -100,6 +100,35 @@ fd_ed25519_verify( void const *  msg,
...
+
+   batch_sz is the size of signatures, pubkeys and shas.
+   batch_sz must be greater than zero.
+
+   See fd_ed25519_verify for more details. */
+
+int
+fd_ed25519_verify_batch_single_msg( uchar const   msg[], /* msg_sz */
+                                    ulong const   msg_sz,
+                                    uchar const   signatures[ static 64 ], /* 64 * batch_sz */
```
Comment (mmcgee-jump) [inline src/ballet/ed25519/fd_ed25519.h:127]: Is the `static 64` correct if it's 64 * batch_sz ?

Comment (0x0ece) [reply src/ballet/ed25519/fd_ed25519.h:127]: static n (by spec) means at least n, so yes, it's correct. it's basically only saying "it must contain at least 1 signature".

File: src/ballet/txn/fd_txn.h (hunk 1 above)
Comment (0x0ece) [reply src/ballet/txn/fd_txn.h:68]: It's in the comment - given the max tx size, you can only fit 12x 64-bit signatures.
I personally think it's better to keep the constants close, so we don't risk to change one but not the other.

File: src/ballet/ed25519/fd_ed25519.h (hunk 2 above)
Comment (mmcgee-jump) [reply src/ballet/ed25519/fd_ed25519.h:127]: :+1:

File: src/ballet/txn/fd_txn.h (hunk 1 above)
Comment (mmcgee-jump) [reply src/ballet/txn/fd_txn.h:68]: Ah, didn't see that thanks

**Pull request merged**